Bottom: 718b7e8505d2deffbcbb5af3e04717491bcf639b
Top:    7f892f4a9f9371062046584a28214c30fb330189
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-10-21 11:54:53 +0100

Refresh of introduce-hacky-code-to-switch

---

diff --git a/src/crosshair.c b/src/crosshair.c
index fa587fe..2c754a2 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -1098,7 +1098,8 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
                                   LINEPOINT_TYPE | ARCPOINT_TYPE,
                                   &ptr1, &ptr2, &ptr3);
       if (ans != NO_TYPE && !TEST_FLAG(SELECTEDFLAG, (LineType *)ptr2)) {
-        gui->endpoint_cursor ();
+        if (gui->endpoint_cursor != NULL)
+          gui->endpoint_cursor ();
         goto common_tail;
       }
 
@@ -1145,13 +1146,15 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
                 break;
             }
 
-            gui->grip_cursor (allowed_directions);
+            if (gui->grip_cursor != NULL)
+              gui->grip_cursor (allowed_directions);
             goto common_tail;
           }
         }
 
     }
-  gui->normal_cursor ();
+  if (gui->normal_cursor != NULL)
+    gui->normal_cursor ();
 
 common_tail:
